### ISSUE TITLE
Parse Server Shutting Down Frequently

### ISSUE THREAD

**Issue by mike3osei** [felixrieseberg/parse-server-azure#9]
After updating my parse server from 2.1.4 on Azure it seems that there is an issue with up-time. Frequently, my parse server is down. Every time it is down I can temporarily fix it by restarting the server at index.js and then everything works fine, however, after 5 minutes it shuts down again. Any ideas on how this issue could be resolved?

**Comment by rmcclus1** [comment]
I am experiencing this same issue. For my case, rest calls to the express app in index.js work fine. Even calls to the Cloud Code main.js work IF they don't deal with the mongodb. Whenever a parse function is ran that requires db interaction, my requests will timeout and no data is served. Why is the express app running fine but the database connection lost? What can be done to keep the connection alive or will azure not be able to perform?

**Comment by rmcclus1** [comment]
There's also a little tidbit to note: I am using parse server 2.1.4 and not this parse server azure library. Should I make the switch to this PS lib since I am using azure? Would it help with the server uptime?

**Comment by felixrieseberg** [comment]
Hey! Sorry that you're having trouble - and thanks for letting us know. We're taking a look right now, chances are that this is totally our fault, given that it sounds a bit like DocumentDB isn't responding quickly enough. I'll keep you posted :memo:!

**Comment by mike3osei** [comment]
Yeah no problem! & sounds good, definitely keep us posted.

**Comment by felixrieseberg** [comment]
Quick update: There are a bunch of fixes that should be deployed over the weekend - again, sorry for your trouble.

**Comment by mike3osei** [comment]
Thanks we appreciate it!

**Comment by rmcclus1** [comment]
Where will those fixes be deployed? Is it with the behind-the-scenes Azure service and will be available immediately or will I need to update certain libraries?

**Comment by felixrieseberg** [comment]
Hopefully, it should all happen behind the scenes. Your app should simply stop restarting 👍

**Comment by rmcclus1** [comment]
Alright thanks! Lastly, the Azure marketplace preconfigured setup for Parse Server comes with Parse's version of PS. Would you recommend sticking with that or switching it out for this parse-server-azure and what would be the benefits/pitfalls of one vs the other?

**Comment by felixrieseberg** [comment]
It's using the same stuff - this repo is just a "I want all the Azure things" collection of tools. You're running the same code 🏄

**Comment by rmcclus1** [comment]
Cool cool. Ok! I'll wait for those fixes to be sent out and report back Monday about the status of the DocuDB responses. Thanks again!

**Comment by felixrieseberg** [comment]
No, thank you for reporting!

**Comment by mike3osei** [comment]
The db is working pretty smooth as of now thank you for patching it up!

**Comment by rmcclus1** [comment]
Yup, it's a lot better now. There were occasional timeouts this morning, but they seem to have been removed as the day went on. Thanks!

Edit: The db connection still has problems holding a connection. Every so often I'll have to go back and restart the server in order to retry the db connection. Is there any way to make the it more stable?

**Comment by mike3osei** [comment]
Yeah, the database was working well for a week or so with no down time. Suddenly today its experiencing downtime again but this time restarting the server doesn't fix it.

**Comment by tanzeelrana** [comment]
My parse server keeps restarting on every invalid credential -_-

![image](https://user-images.githubusercontent.com/2520028/27009420-deb273b8-4e5b-11e7-9905-ce1c5dc89e75.png)
